plugin/storage/kafka/options.go

```diff
@@ -94,3 +181,37 @@ func (opt *Options) InitFromViper(v *viper.Viper) {
 func stripWhiteSpace(str string) string {
 	return strings.Replace(str, " ", "", -1)
 }
+
+// getCompressionLevel to get compression level from compression type
+func getCompressionLevel(mode string, compressionLevel int) (int, error) {
+	compressionModeData := compressionModes[mode]
+
+	if compressionLevel == defaultCompressionLevel {
+		return compressionModeData.defaultCompressionLevel, nil
+	}
+
+	if compressionModeData.minCompressionLevel > compressionLevel || compressionModeData.maxCompressionLevel < compressionLevel {
+		return 0, fmt.Errorf("compression level is not within valid range")
+	}
+
+	return compressionLevel, nil
+}
+
+//getCompressionModes maps input modes to sarama CompressionCodec
+func getCompressionModes(mode string) (sarama.CompressionCodec, error) {
```

Comment:
s/getCompressionModes/getCompressionMode/ (it returns a singular value, not plural)

Reply:
fixed